x/concentrated-liquidity/swaps.go

```diff
@@ -362,13 +362,27 @@ func (k Keeper) computeOutAmtGivenIn(
 			if err != nil {
 				return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, sdk.Dec{}, err
 			}
+			// Update the swapState's tick with the tick we retrieved liquidity from
+			swapState.tick = swapStrategy.UpdateTickAfterCrossing(nextTick)
```

Comment:
Note to reviewer: this is the key realization and fix to the original bug as opposed to tick initialization as originally thought

Once the general change is accepted, I will improve the abstraction by moving it inside the `swapCrossTickLogic` method

Reply:
This makes sense per discussions!